File: unsolved/2108.cpp
#include <iostream>
#include <cmath>
#include <algorithm>
using namespace std;
#define endl '\n'

int main(){
    int N;
    cin >> N;
    int nums[N];
    int sum = 0;
    for(int i = 0; i < N; i++){
        cin >> nums[i];
        sum += nums[i];
    }
    sort(nums, nums+N);
    cout << floor(sum / N + 0.5) << endl;
    cout << nums[(N-1)/2];

    return 0;
}

File: unsolved/2108.py
import sys
import math
N = int(sys.stdin.readline())

nums = []
how_many_nums = {}
most_values = []
most_times = 0
for _ in range(N):
    num = int(sys.stdin.readline())
    nums.append(num)
    if num not in how_many_nums:
        how_many_nums[num] = 1
    else:
        how_many_nums[num] += 1

for value in how_many_nums:
    if most_times < how_many_nums[value]:
        most_times = how_many_nums[value]
        most_values = [value]
    elif most_times == how_many_nums[value]:
        most_values.append(value)

nums.sort()
most_values.sort(reverse=True)

print(math.floor(sum(nums)/N + 0.5))
print(nums[int((N-1)/2)])
if len(most_values) == 1:
    print(most_values[0])
else:
    print(most_values[-2])
print(nums[-1] - nums[0])
